### ISSUE TITLE
Doubt in code in the model/faster_rcnn/resnet.py file

### ISSUE THREAD

**Issue by jiteshm17** [jwyang/faster-rcnn.pytorch#771]
Hi, I actually have a few doubts in the resnet.py file located in the `lib/model/faster_rcnn` directory starting from this particular [line](https://github.com/tdchaitanya/MMTOD/blob/3a438121c663c2af00da46fed3b467c2012c61ed/lib/model/faster_rcnn/resnet.py#L249)  

My first doubt is, is there any reason you are freezing only some specific layers and not all ?
Second, in the `def train` function below, is setting the `RCNN_base` to eval() same as setting all the layers having the property of `requires_grad` as `False` ? And according to my understanding, according to the next 2 lines, I guess you are training only the set of 5th and 6th layers. Please let me know if my understanding is correct.

**Comment by loolzaaa** [comment]
> My first doubt is, is there any reason you are freezing only some specific layers and not all ?

I think the reason is that network need to be trained at least some conv layers, not only classifier.

Also, i think there is an error in code. If you change cfg.RESNET.FIXED_BLOCKS parameter to >1, then train method doesn't turn on eval mode in RCNN_base[5...6].

I suggest this option in _init_modules method:
```python
assert (0 <= cfg.RESNET.NUM_FREEZE_BLOCKS < 4)
for i in range(4, cfg.RESNET.NUM_FREEZE_BLOCKS + 4):
    for p in self.RCNN_base[i].parameters(): p.requires_grad=False

def batch_norm_grad_turn_off(module):
    if isinstance(module, nn.modules.batchnorm._BatchNorm):
        for p in module.parameters(): p.requires_grad = False

self.RCNN_base(batch_norm_grad_turn_off)
self.RCNN_top(batch_norm_grad_turn_off)
```
and this in overrided train method:
```python
def train(self, mode=True):
    # Override train so that the training mode is set as we want
    super().train(mode)
    if mode:
        # Set fixed blocks to be in eval mode
        self.RCNN_base.eval()
        for i in range(4 + cfg.RESNET.NUM_FREEZE_BLOCKS, 7):
            self.RCNN_base[i].train()

        def batch_norm_eval_turn_on(module):
            if isinstance(module, nn.modules.batchnorm._BatchNorm):
                module.eval()

        self.RCNN_base.apply(batch_norm_eval_turn_on)
        self.RCNN_top.apply(batch_norm_eval_turn_on)
```
> I guess you are training only the set of 5th and 6th layers

+ RCNN_top set + RCNN_bbox + RCNN_loss
